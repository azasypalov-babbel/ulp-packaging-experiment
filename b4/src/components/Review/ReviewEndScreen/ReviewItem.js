import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../lib/cxHelper';
import { getFirstCorrectSolution } from '@lessonnine/babbel-markup-helper.js';

const ReviewItem = ({ item, onItemClick, ...props }) => (
  <div
    className={cx('review-item')}
    onClick={() => onItemClick(item)}
    {...props}
  >
    <div className={cx('review-item__icon')} />
    <div
      className={cx('review-item__text')}
      dangerouslySetInnerHTML={{ __html: getFirstCorrectSolution(item.learnLanguageText) }}
    />
  </div>
);

ReviewItem.propTypes = {
  item: PropTypes.shape({
    learnLanguageText: PropTypes.string.isRequired
  }).isRequired,
  onItemClick: PropTypes.func.isRequired
};

export default ReviewItem;
