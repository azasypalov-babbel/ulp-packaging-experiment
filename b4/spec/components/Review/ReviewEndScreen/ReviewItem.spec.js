import React from 'react';
import ReviewItem from '../../../../src/components/Review/ReviewEndScreen/ReviewItem';
import { shallow } from 'enzyme';

const defaultProps = {
  item: {
    learnLanguageText: 'die Straße',
    sound: { id: '100afffb3' }
  },
  onItemClick: () => {}
};

describe('<ReviewItem />', () => {
  test('it renders', () => {
    const component = shallow(
      <ReviewItem {...defaultProps} />
    );
    expect(component).toMatchSnapshot();
  });

  test('it renders with item containing html tags', () => {
    const props = {
      item: {
        learnLanguageText: '<i>die Straße</i>',
        sound: { id: '100afffb3' }
      },
      onItemClick: () => {}
    };
    const component = shallow(
      <ReviewItem {...props} />
    );

    expect(component.html()).toMatch('<i>die Straße</i>');
  });
});
