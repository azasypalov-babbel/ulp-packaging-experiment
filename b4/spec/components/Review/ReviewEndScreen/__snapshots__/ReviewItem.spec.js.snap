// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewItem /> it renders 1`] = `
<div
  className="loy-review-item"
  onClick={[Function]}
>
  <div
    className="loy-review-item__icon"
  />
  <div
    className="loy-review-item__text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "die StraÃŸe",
      }
    }
  />
</div>
`;
